File: java/src/model/VarExpr.java
package model;
import utils.*;

public class VarExpr implements Expression{
	private String name;
	public VarExpr(String name){
		this.name=name;
	}
	
	public int evaluate(ISymbolTable<String,Integer> s,IHeap<Integer,Integer> hp){
		if(s.contains(name))
			return s.getValue(name);
		else throw new InterpretorException("no such variable");
	}
	
	public String toString() {
		return name;
	}
}

File: C#/ToyLanguageC/model/VarExpr.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ToyLanguageC.utils;

namespace ToyLanguageC.model
{
    class VarExpr : Expression
    {
        private string var;

        public VarExpr(string var)
        {
            this.var = var;
        }

        public int evaluate(SymbolTable<string, int> sym)
        {
            return sym.getValue(var);
        }

        public string toString()
        {
            return var;
        }
    }
}
